src/fides/api/service/messaging/messaging_crud_service.py

```diff
@@ -81,31 +87,261 @@ def get_messaging_config_by_key(db: Session, key: FidesKey) -> MessagingConfigRe
+    )
+    if not db_messaging_template:
+        raise MessagingConfigNotFoundException(
+            f"No messaging template found with id {template_id}"
+        )
+    _validate_overlapping_templates(
+        db,
+        db_messaging_template.type,
+        template_update_body.properties,
+        template_update_body.is_enabled,
+        template_id,
+    )
+
+    data = {
+        "content": template_update_body.content,
+        "is_enabled": template_update_body.is_enabled,
+    }
+    if template_update_body.properties:
+        data["properties"] = [
+            {"id": property_id} for property_id in template_update_body.properties
+        ]
+
+    return db_messaging_template.update(db=db, data=data)
+
+
+def create_messaging_template(
+    db: Session,
+    template_type: str,
+    template_create_body: MessagingTemplateWithPropertiesBodyParams,
+) -> Optional[MessagingTemplate]:
+    if template_type not in DEFAULT_MESSAGING_TEMPLATES:
+        raise MessagingConfigValidationException(
+            f"Messaging template type {template_type} is not supported."
+        )
+    _validate_overlapping_templates(
+        db,
+        template_type,
+        template_create_body.properties,
+        template_create_body.is_enabled,
+        None,
+    )
+
+    data = {
+        "content": template_create_body.content,
+        "is_enabled": template_create_body.is_enabled,
+        "type": template_type,
+    }
+    if template_create_body.properties:
+        data["properties"] = [
+            {"id": property_id} for property_id in template_create_body.properties
+        ]
+    return MessagingTemplate.create(db=db, data=data)
+
+
+def delete_template_by_id(db: Session, template_id: str) -> None:
+    logger.info("Finding messaging config with id '{}'", template_id)
+    messaging_template: Optional[MessagingTemplate] = MessagingTemplate.get(
+        db, object_id=template_id
+    )
+    if not messaging_template:
+        raise MessagingConfigNotFoundException(
+            f"No messaging template found with id {template_id}"
+        )
+    templates_with_type = (
+        MessagingTemplate.query(db=db)
+        .filter(MessagingTemplate.type == messaging_template.type)
+        .all()
+    )
+    if len(templates_with_type) <= 1:
+        raise MessagingConfigValidationException(
+            f"Messaging template with id {template_id} cannot be deleted because it is the only template with type {messaging_template.type}"
+        )
+    logger.info("Deleting messaging config with id '{}'", template_id)
+    messaging_template.delete(db)
+
+
+def get_template_by_id(db: Session, template_id: str) -> MessagingTemplate:
+    messaging_template: Optional[MessagingTemplate] = MessagingTemplate.get(
+        db, object_id=template_id
+    )
+    if not messaging_template:
+        raise MessagingConfigNotFoundException(
+            f"No messaging template found with id {template_id}"
+        )
+    return messaging_template
```

Comment:
You might be able to make use of this new method in other locations on this page -

Reply:
great idea 💯

Reply:
you might be able to use in update_messaging_template too?